packages/core/src/database/index.ts

```diff
@@ -1115,8 +1008,116 @@ export const createDatabase = async ({
           });
         }
       }, common.options.databaseHeartbeatInterval);
+    },
+    async getListenConnection() {
+      const trigger = "status_trigger";
+      const notification = `${preBuild.namespace}_status_notify()`;
+      const channel = `${preBuild.namespace}_status_channel`;
+
+      await this.wrap({ method: "getListenConnection" }, async () => {
+        await sql
+          .raw(`
+      CREATE OR REPLACE FUNCTION ${notification}
+      RETURNS trigger
+      LANGUAGE plpgsql
+      AS $$
+      BEGIN
+      NOTIFY ${channel};
+      RETURN NULL;
+      END;
+      $$;`)
+          .execute(qb.internal);
+
+        await sql
+          .raw(`
+      CREATE OR REPLACE TRIGGER ${trigger}
+      AFTER INSERT OR UPDATE OR DELETE
+      ON "${preBuild.namespace}"._ponder_status
+      FOR EACH STATEMENT
+      EXECUTE PROCEDURE ${notification};`)
+          .execute(qb.internal);
+      });
```

Comment:
My instinct is that these should happen in the same spot that we create the `_ponder_status` table (if not exists).

Reply:
The problem is that the function that creates the status table (`migrate()`) isn't always run when `getListenConnection()` is run. For example the `serve` command. We might be able to come up with a better solution than I have now but that was my original reason for making this change.